text/0031-transactions-with-contingent-inputs.md

```diff
@@ -0,0 +1,191 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+In a MobileCoin transaction, an input consists of a collection of `TxOut`'s in the ledger, one of which is truly being spent and others of which are just being mixed in. This ring,
+and the proofs of membership, appear in the `TxIn`, in the `TxPrefix`. A transaction produces several new `TxOut`'s, and these outputs appear in the `TxPrefix`.
+
+The signature part of a transaction contains several things:
+* A `RingMLSAG` for each `TxIn`
+* A `pseudo_output_commitment` for each `TxIn`
+* A range proof for each `pseudo_output_commitment`, relative to the generator for this token id.
+
+Transaction validation ensures that the sum of outputs minus the sum of pseudo-outputs is the zero point on the elliptic curve, (including the implicit fee output)
+which ensures that the transaction is valid.
+
+This feature contains two discrete steps:
+* Allowing mixed transactions. This requires that individual outputs and pseudo-outputs in a transaction can be labeled with token ids.
+  (The MCIP #25 token id, added to the tx prefix, is now interpreted as the fee token id.)
+  When validating range proofs, each individual output and pseudo-output has its range proof checked relative to its specified token id.
+  No other changes are required to support mixed transactions.
+* Allowing contingent inputs in a transaction. A contingent input is a `TxIn` together with an additional `ContingentInputRules` object.
+  The `RingMLSAG` for a contingent input signs over (a hash of) the `ContingentInputRules`. When validating a Tx with contingent input rules,
+  those rules must be satisfied by the transaction as a whole for the transaction to be valid.
+  Examples of rules include: "a particular output must appear in the `TxPrefix`", "the tombstone block may not exceed" a certain value.
+  Because the `RingMLSAG` signs over the `ContingentInputRules`, these rules cannot be tampered with once the `RingMLSAG` is signed.
+
+A "signed contingent input" is a payload consisting of:
+* A `TxIn`, with `ContingentInputRules`.
+* A `pseudo_output_commitment` for this ring.
+* A `RingMLSAG` signing this.
+* A token id, value, and blinding factor for the `pseudo_output_commitment`.
+* A token id, value, and blinding factor for any output in the `ContingentInputRules`.
+
+Anyone who owns a `TxOut` can use it to create a `SignedContingentInput`. The signed contingent input can be thought of as an offer
+to trade "this for that" -- you are willing to give away your `TxOut` if someone produces for you the `TxOut` that you specify.
+Once you sign the contingent input, you can give it away, and anyone who has it can fulfill your request to trade.
+
+Giving away the signed contingent input does not reveal your public address, nor which `TxOut` in the ledger you actually own.
+It does reveal the token id and value of your pseudo-output, but this is unavoidable in any such scheme.
```

Comment:
Can you describe a bit more here? Or maybe this will go in the reference-level explanation - I'm trying to imagine the "shared secrets" involved